--- src/torrent/data/file.cc	2015-08-08 16:01:49.000000000 +0100
+++ src/torrent/data/file.cc	2016-08-21 22:11:53.000144395 +0100
@@ -168,13 +168,14 @@ File::resize_file() {
   if (m_size == SocketFile(m_fd).size())
     return true;
 
-  // For now make it so that the fallocate flag indicates if we want
-  // to do potentially blocking allocation, while FS supported
-  // non-blocking allocation is done always.
-  int flags = SocketFile::flag_fallocate;
+  int flags = 0;
 
-  if (m_flags & flag_fallocate)
+  // Set FS supported non-blocking allocation flag and potentially
+  // blocking allocation flag if fallocate flag is set.
+  if (m_flags & flag_fallocate) {
+    flags |= SocketFile::flag_fallocate;
     flags |= SocketFile::flag_fallocate_blocking;
+  }
 
   return SocketFile(m_fd).set_size(m_size, flags);
 }
