--- orig/src/core/curl_stack.h	2009-11-12 09:03:44.000000000 +0100
+++ rtorrent-0.8.6/src/core/curl_stack.h	2011-06-03 12:07:25.000000000 +0200
@@ -91,2 +91,5 @@
 
+  int                 dns_cache_timeout() const              { return m_dns_cache_timeout; }
+  void                set_dns_cache_timeout(int t)           { m_dns_cache_timeout = t; }
+
   const std::string&  user_agent() const                     { return m_userAgent; }
@@ -132,2 +138,3 @@
 
+  int                 m_dns_cache_timeout;
   std::string         m_userAgent;
--- orig/src/core/curl_stack.cc	2009-11-12 09:03:44.000000000 +0100
+++ rtorrent-0.8.6/src/core/curl_stack.cc	2011-06-03 12:09:10.000000000 +0200
@@ -52,2 +52,3 @@
   m_handle((void*)curl_multi_init()),
+  m_dns_cache_timeout(60),
   m_active(0),
@@ -154,2 +156,3 @@
 
+  curl_easy_setopt(get->handle(), CURLOPT_DNS_CACHE_TIMEOUT, m_dns_cache_timeout);
   if (!m_httpProxy.empty())

--- orig/src/command_network.cc	2011-04-23 14:27:56.000000000 +0200
+++ rtorrent-0.8.8/src/command_network.cc	2011-06-05 12:51:36.000000000 +0200
@@ -677,2 +680,5 @@
 
+  CMD2_ANY         ("network.http.dns_cache_timeout",     std::bind(&core::CurlStack::dns_cache_timeout, httpStack));
+  CMD2_ANY_VALUE_V ("network.http.dns_cache_timeout.set", std::bind(&core::CurlStack::set_dns_cache_timeout, httpStack, std::placeholders::_2));
+
   CMD2_ANY         ("network.http.capath",              std::bind(&core::CurlStack::http_capath, httpStack));
