--- rel-0.9.4/src/ui/download.cc	2014-05-14 16:30:51.000000000 +0200
+++ rtorrent-0.9.4/src/ui/download.cc	2015-08-11 03:31:14.000000000 +0200
@@ -171,6 +171,6 @@
   element->push_back("");
   element->push_column("Memory usage:",     te_command("cat=$convert.mb=$pieces.memory.current=,\" MB\""));
-  element->push_column("Max memory usage:", te_command("cat=$convert.mb=$pieces.memory.max=,\" MB\""));
-  element->push_column("Free diskspace:",   te_command("cat=$convert.mb=$d.free_diskspace=,\" MB\""));
+  element->push_column("Max memory usage:", te_command("cat=$convert.mb=$pieces.memory.max=,\" MB = \",$convert.xb=$pieces.memory.max="));
+  element->push_column("Free diskspace:",   te_command("cat=$convert.mb=$d.free_diskspace=,\" MB = \",$convert.xb=$d.free_diskspace="));
   element->push_column("Safe diskspace:",   te_command("cat=$convert.mb=$pieces.sync.safe_free_diskspace=,\" MB\""));
 
@@ -192,7 +192,7 @@
 
   element->push_back("");
-  element->push_column("Upload:",           te_command("cat=$convert.kb=$d.up.rate=,\" KB / \",$convert.xb=$d.up.total="));
-  element->push_column("Download:",         te_command("cat=$convert.kb=$d.down.rate=,\" KB / \",$convert.xb=$d.down.total="));
-  element->push_column("Skipped:",          te_command("cat=$convert.kb=$d.skip.rate=,\" KB / \",$convert.xb=$d.skip.total="));
+  element->push_column("Upload:",           te_command("cat=$convert.xb=$d.up.rate=,\" / \",$convert.xb=$d.up.total="));
+  element->push_column("Download:",         te_command("cat=$convert.xb=$d.down.rate=,\" / \",$convert.xb=$d.down.total="));
+  element->push_column("Skipped:",          te_command("cat=$convert.xb=$d.skip.rate=,\" / \",$convert.xb=$d.skip.total="));
   element->push_column("Preload:",          te_command("cat=$pieces.preload.type=,\" / \",$pieces.stats_preloaded=,\" / \",$pieces.stats_preloaded="));
