diff --git a/src/rpc/scgi_task.h b/src/rpc/scgi_task.h
index e75e9e1..c36de13 100644
--- a/src/rpc/scgi_task.h
+++ b/src/rpc/scgi_task.h
@@ -51,7 +51,7 @@ class SCgiTask : public torrent::Event {
 public:
   static const unsigned int default_buffer_size = 2047;
   static const          int max_header_size     = 2000;
-  static const          int max_content_size    = (2 << 20);
+  static const          int max_content_size    = (2 << 23);
 
   SCgiTask() { m_fileDesc = -1; }
 
