--- src/data/memory_chunk.cc
+++ src/data/memory_chunk.cc
@@ -69,7 +69,7 @@
   if (page_align() >= m_pagesize)
     throw internal_error("MemoryChunk::MemoryChunk(...) received an page alignment >= page size");
 
-  if ((ptrdiff_t)ptr % m_pagesize)
+  if ((std::ptrdiff_t)ptr % m_pagesize)
     throw internal_error("MemoryChunk::MemoryChunk(...) is not aligned to a page");
 }
 
--- src/torrent/data/block.h
+++ src/torrent/data/block.h
@@ -40,6 +40,7 @@
 #include <vector>
 #include <torrent/common.h>
 #include <torrent/data/block_transfer.h>
+#include <cstdlib>
 
 namespace torrent {
 
--- src/torrent/data/block_transfer.h
+++ src/torrent/data/block_transfer.h
@@ -39,6 +39,7 @@
 
 #include <torrent/common.h>
 #include <torrent/data/piece.h>
+#include <cstdlib>
 
 namespace torrent {
 
